glambie/data/read_and_edit_bucket_csv_files.py

```diff
@@ -0,0 +1,343 @@
+                                'glacier_area_reference'][0] for i in range(len(interpolated_data_frame))]
+                    else:
+                        print('Some different reference areas')
+
+                    if all(i == submission_data_frame['glacier_area_observed'][0] for i in submission_data_frame[
+                            'glacier_area_observed']):
+                        interpolated_data_frame[
+                            'glacier_area_observed'] = [submission_data_frame[
+                                'glacier_area_observed'][0] for i in range(len(interpolated_data_frame))]
+                    else:
+                        print('Some different observed areas')
+
+                    if all(i == submission_data_frame.remarks[0] for i in submission_data_frame.remarks) or \
+                            all(np.isnan(submission_data_frame.remarks)):
+                        interpolated_data_frame['remarks'] = [submission_data_frame.remarks[0]
+                                                              for i in range(len(interpolated_data_frame))]
+                    else:
+                        print('Some different remarks')
+
+                    interpolated_data_frame['user_group'] = [submission_data_frame['user_group'][0]
+                                                             for i in range(len(interpolated_data_frame))]
+                    interpolated_data_frame['start_date_fractional'] = start_date_fractional
+                    interpolated_data_frame['end_date_fractional'] = end_date_fractional
+                    interpolated_data_frame['date'] = interpolated_data_frame['start_date']
+                    interpolated_data_frame['date_fractional'] = start_date_fractional
+                    interpolated_data_frame['region_id'] = [submission_data_frame['region_id'][0]
+                                                            for i in range(len(interpolated_data_frame))]
+
+                    submission_data_frame = interpolated_data_frame.copy()
+
+                else:
+                    # If it is not a gravimetry file, then the data it contains is invalid.
+                    file_check_dataframe.loc[
+                        file_check_dataframe.local_filepath.__eq__(file.local_filepath),
+                        'reason_for_edit'] = 'Non-gravimetry file with gaps in data - inspect manually'
+
+            # Save out to same path as original file
+            submission_data_frame.to_csv(file.local_filepath)
+
+        if not file.nodata_check_satisfied:
+            submission_data_frame = pd.read_csv(file.local_filepath)
+
+            if 'm' in submission_data_frame.unit.values[0]:
+                # delete rows with change values > +/-100 - these numbers need some thought
+                submission_data_frame.drop(submission_data_frame[abs(
+                    submission_data_frame.glacier_change_observed) > 100].index, inplace=True)
```

Comment:
Should 100 and 10000 be consts? They are used in multiple places

Reply:
Done!